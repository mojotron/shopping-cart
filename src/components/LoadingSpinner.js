import "../styles/LoadingSpinner.css";
import spinner from "../images/loading.svg";

const LoadingSpinner = () => (
  <div className="LoadingSpinner">
    <img
      className="LoadingSpinner__spinner"
      src={spinner}
      alt="loading spinner"
    />
  </div>
);

export default LoadingSpinner;
