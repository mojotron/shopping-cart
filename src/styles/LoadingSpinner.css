.LoadingSpinner {
  height: 5rem;
  width: 5rem;
}
.LoadingSpinner__spinner {
  animation: spin 3s linear infinite;
}
@keyframes spin {
  0% {
    transform: rotate(360deg);
  }
  100% {
    transform: rotate(0deg);
  }
}
