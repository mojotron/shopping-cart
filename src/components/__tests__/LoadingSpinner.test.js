import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import LoadingSpinner from "../LoadingSpinner";

describe("LoadingSpinner component", () => {
  test("render LoadingSpinner snapshot", () => {
    const { container } = render(<LoadingSpinner />);
    expect(container).toMatchSnapshot();
  });

  test("has image", () => {
    render(<LoadingSpinner />);
    expect(screen.getByRole("img")).toBeInTheDocument();
  });

  test("image has spinning class", () => {
    render(<LoadingSpinner />);
    expect(screen.getByRole("img")).toHaveClass("LoadingSpinner__spinner");
  });
});
