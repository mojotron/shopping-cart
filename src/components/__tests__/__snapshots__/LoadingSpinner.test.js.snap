// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner component render LoadingSpinner snapshot 1`] = `
<div>
  <div
    class="LoadingSpinner"
  >
    <img
      alt="loading spinner"
      class="LoadingSpinner__spinner"
      src="loading.svg"
    />
  </div>
</div>
`;
